caster.go

```diff
@@ -0,0 +1,41 @@
+package qkit
+
+import "encoding/json"
+
+// Converts any type to a given type. If conversion fails, it returns the zero value of the given type.
+func Cast[T any](val any) T {
+	if val, ok := val.(T); ok {
+		return val
+	}
+	var zero T
+	return zero
+}
+
+// Converts any type to a given struct type.
```

Comment:
Name of function is confusing, ToStruct means that output is struct but the type parameter is `any`. So I can call it like: ToStruct[string](struct{....}) this always return empty value. Also i can pass ToStruct[map[string]any] so it does not make sense.

what is the usecase for this function?  to receive an `object1` and produce `object2` based on their json tags or reflection on field names ?

the current code i see it as a shorthand for `FromBytes[B](ToBytes(x))`,

Reply:
Yes that without type assertion, and to convert something like a `map to a struct`. Do you have another name in mind, because in this case it would be nice to use a shorthand instead of `FromBytes[B](ToBytes(x))`.

Reply:
I've renamed it to CastPartial and cleaned up the code, please do let me know if you have any other suggestions